homeassistant/components/delijn/sensor.py

```diff
@@ -0,0 +1,135 @@
+        line = Passages(hass.loop,
+                        stop_id,
+                        number_of_departures,
+                        api_key,
+                        session)
+        await line.get_passages()
+        if line.passages is None:
+            _LOGGER.error("No data recieved from De Lijn")
+            return
+        sensors.append(DeLijnPublicTransportSensor(line, name))
+
+    async_add_entities(sensors, True)
+
+
+class DeLijnPublicTransportSensor(Entity):
+    """Representation of a Ruter sensor."""
+
+    def __init__(self, line, name):
+        """Initialize the sensor."""
+        self.line = line
+        self._attributes = {}
+        self._name = name
+        self._state = None
+
+    async def async_update(self):
+        """Get the latest data from the De Lijn API."""
+        await self.line.get_passages()
+        if self.line.passages is None:
+            _LOGGER.error("No data recieved from De Lijn")
+            return
+        try:
+            attributes = {}
+            first = self.line.passages[0]
+            local_timezone = pytz.timezone("Europe/Brussels")
+            if first['due_at_rt'] is not None:
+                first_passage_naive = dt_util.parse_datetime(
+                    first['due_at_rt'])
+            else:
+                first_passage_naive = dt_util.parse_datetime(
+                    first['due_at_sch'])
+            first_passage_local = local_timezone.localize(first_passage_naive)
+            first_passage_utc = dt_util.as_utc(first_passage_local)
+            self._state = first_passage_utc.isoformat()
+            self._name = first['stopname']
+            attributes['stopname'] = first['stopname']
+            attributes['line_number_public'] = first['line_number_public']
+            attributes['line_transport_type'] = first['line_transport_type']
+            attributes['final_destination'] = first['final_destination']
+            attributes['due_at_schedule'] = first['due_at_sch']
+            attributes['due_at_realtime'] = first['due_at_rt']
+            attributes['due_in_min'] = first['due_in_min']
+            attributes['next_passages'] = self.line.passages
+            self._attributes = attributes
+        except (KeyError, IndexError) as error:
+            _LOGGER.debug("Error getting data from De Lijn, {}".format(error))
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return DEVICE_CLASS_TIMESTAMP
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        return 'mdi:bus'
+
+    @property
+    def device_state_attributes(self):
+        """Return attributes for the sensor."""
+        self._attributes[ATTR_ATTRIBUTION] = ATTRIBUTION
```

Comment:
Include the attribution already when creating the dict instance attribute in init method.

Reply:
Added the attribution to the init method in the latest commit.

Reply:
We can remove this if we update instead of replacing the attributes.

Reply:
Done, updated in the latest commit